### PR TITLE
LLT-5325 Add OpenWRT linker backport v4.3

### PR DESCRIPTION
**Pull request by matislovas**
Same as https://github.com/NordSecurity/libtelio/pull/624

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] changelog.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

**Review by gytsto — DISMISSED** [review]
+1

File: Cargo.lock [hunk 1]
```diff
@@ -119,7 +119,7 @@ version = "1.0.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
```
Comment (Jauler) [inline Cargo.lock:119]: was the update of the packages here intentional?

File: linkers/d-x86_64-openwrt-linux-musl [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+# This forces export of needed binding for Java
+LINKER=$(basename $0)
+LINKER=${LINKER#"d-"}
+ARGS=$@
+
+while (( "$#" )); do
+	if [[ $1 == *"-Wl,--version-script="* && ! $ARGS =~ -o.*librustls_platform_verifier.* ]]; then
+		VERSION=${1#"-Wl,--version-script="}
+		SED_ARGS='s/global:/global:\n    Java*;\n    JNI_OnLoad*;\n    _wrap*;\n/'
```
Comment (Jauler) [inline linkers/d-x86_64-openwrt-linux-musl:12]: OpenWRT probably will not ever need Java symbols :) 
```suggestion
		SED_ARGS='s/global:/global:\n    _wrap*;\n/'
```

File: linkers/d-aarch64-openwrt-linux-musl [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# This forces export of needed binding for Java
+LINKER=$(basename $0)
+LINKER=${LINKER#"d-"}
+ARGS=$@
+
+while (( "$#" )); do
+	if [[ $1 == *"-Wl,--version-script="* && ! $ARGS =~ -o.*librustls_platform_verifier.* ]]; then
+		VERSION=${1#"-Wl,--version-script="}
+		SED_ARGS='s/global:/global:\n    Java*;\n    JNI_OnLoad*;\n    _wrap*;\n/'
```
Comment (Jauler) [inline linkers/d-aarch64-openwrt-linux-musl:12]: OpenWRT probably will not ever need Java symbols :)
```suggestion
		SED_ARGS='s/global:/global:\n    _wrap*;\n/'
```

File: linkers/d-x86_64-openwrt-linux-musl (hunk 2 above)
Comment (matislovas) [reply linkers/d-x86_64-openwrt-linux-musl:12]: Just copied out from the others, but yeah, will adjust

File: Cargo.lock (hunk 1 above)
Comment (matislovas) [reply Cargo.lock]: Yeah, a leftover, deleted.

**Review by Jauler — APPROVED** [review]
+1.0

**Pull request merged**